### PR TITLE
fix(iot-service): Fix bug where twin metadata isn't given to user even when it was received from service

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Example twin:
```
{
    "deviceId": "devA",
    "etag": "AAAAAAAAAAc=", 
    "status": "enabled",
    "statusReason": "provisioned",
    "statusUpdateTime": "0001-01-01T00:00:00",
    "connectionState": "connected",
    "lastActivityTime": "2015-02-30T16:24:48.789Z",
    "cloudToDeviceMessageCount": 0, 
    "authenticationType": "sas",
    "x509Thumbprint": {     
        "primaryThumbprint": null, 
        "secondaryThumbprint": null 
    }, 
    "version": 2, 
    "tags": {
        "$etag": "123",
        "deploymentLocation": {
            "building": "43",
            "floor": "1"
        }
    },
    "properties": {
        "desired": {
            "telemetryConfig": {
                "sendFrequency": "5m"
            },
            "$metadata" : {...}, // <----- was missing
            "$version": 1
        },
        "reported": {
            "telemetryConfig": {
                "sendFrequency": "5m",
                "status": "success"
            },
            "batteryLevel": 55,
            "$metadata" : {...}, // <----- was missing
            "$version": 4
        }
    }
}
```

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinProperties.java [hunk 1]
```diff
@@ -130,18 +130,10 @@
         if (desired != null)
         {
             this.desired = TwinCollection.createFromRawCollection(desired);
-            if (desired.getVersion() != null)
-            {
-                this.desired.setVersion(desired.getVersion());
-            }
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinProperties.java:136]: The above line already does this, so no need to manually assign the version here, too

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinCollection.java [hunk 2]
```diff
@@ -151,6 +151,11 @@ public TwinCollection(TwinCollection collection) {
         }
     }
 
+    final Map<String, TwinMetadata> getMetadataMap()
```
Comment (brycewang-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinCollection.java:154]: Add javadoc on this one?

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinCollection.java [hunk 3]
```diff
@@ -413,6 +418,10 @@ public final TwinMetadata getTwinMetadata() {
         return this.twinMetadata;
     }
 
+    final void setTwinMetadata(TwinMetadata twinMetadata) {
```
Comment (brycewang-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinCollection.java:421]: javadoc?

**Comment by brycewang-microsoft** [comment]
In the class `Twin`, for the method `fromJson()` (starting from L74), can we reorder the DesiredProperty-related lines and ReportedProperty-related lines? Currently these 2 parts are interlacing with each other.

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/TwinClientTests.java [hunk 4]
```diff
@@ -92,6 +95,9 @@ public void testPatchTwin() throws IOException, GeneralSecurityException, IotHub
         // assert
         assertTrue(TwinCommon.isPropertyInTwinCollection(twin.getDesiredProperties(), expectedDesiredPropertyKey, expectedDesiredPropertyValue));
         assertNotNull(twin.getDesiredProperties().getVersion());
+        assertNotNull(twin.getDesiredProperties().getTwinMetadata());
```
[the remark this answers is not in the document]
Comment (timtay-microsoft) [reply iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/TwinClientTests.java:98]: Sure thing

**Pull request merged**